modules/apps/configuration-admin/configuration-admin-api/src/main/java/com/liferay/configuration/admin/display/ConfigurationScreenAlertProvider.java

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.configuration.admin.display;
+
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+
+import java.io.Serializable;
+
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * @author Alejandro Tardín
+ */
+public interface ConfigurationScreenAlertProvider {
+
+	public Alert getAlert(
+		ExtendedObjectClassDefinition.Scope scope, Serializable scopePK);
+
+	public interface Alert {
+
+		public String getType();
+
+		public void render(
+				HttpServletRequest httpServletRequest,
+				HttpServletResponse httpServletResponse)
+			throws Exception;
```

Comment:
Could we get away with just asking for a string? Between the type and the message we should be covered, right? Or is there another use-case to consider?

Reply:
Hi @drewbrokke,

We were considering to add a button to try again if the download fails.

We can just use a string as long as it is rendered as HTML. Then we would be covered for those potential use cases.

Reply:
But if we decide to support HTML I think it is more clear to use a render method instead of a String. What do you think?

Reply:
That makes sense. If that's the case, it may just be simpler to include a dynamic-include tag on the edit configuration form. Then you can render whatever is needed with no new api. Thoughts?

Reply:
I can send you a pull with your business logic applied to a dynamic include if you'd like.

Reply:
I thought about doing it that way too. I think it's a good idea, go for it. Or I can do it myself if you prefer so.

Thanks!